src/main/java/de/tum/in/www1/artemis/service/connectors/athena/AthenaSubmissionSendingService.java

```diff
@@ -0,0 +1,104 @@
+import de.tum.in.www1.artemis.exception.NetworkingError;
+import de.tum.in.www1.artemis.repository.TextSubmissionRepository;
+import de.tum.in.www1.artemis.service.dto.athena.TextExerciseDTO;
+import de.tum.in.www1.artemis.service.dto.athena.TextSubmissionDTO;
+
+/**
+ * Service for sending submissions to the Athena service for further processing
+ * so that Athena can later give feedback suggestions on new submissions.
+ */
+@Service
+@Profile("athena")
+public class AthenaSubmissionSendingService {
+
+    private final Logger log = LoggerFactory.getLogger(AthenaSubmissionSendingService.class);
+
+    @Value("${artemis.athena.url}")
+    private String athenaUrl;
+
+    private final TextSubmissionRepository textSubmissionRepository;
+
+    private final AthenaConnector<RequestDTO, ResponseDTO> connector;
+
+    public AthenaSubmissionSendingService(TextSubmissionRepository textSubmissionRepository, @Qualifier("athenaRestTemplate") RestTemplate athenaRestTemplate) {
+        this.textSubmissionRepository = textSubmissionRepository;
+        connector = new AthenaConnector<>(log, athenaRestTemplate, ResponseDTO.class);
+    }
+
+    private static class RequestDTO {
+
+        public TextExerciseDTO exercise;
+
+        public List<TextSubmissionDTO> submissions;
+
+        RequestDTO(@NotNull TextExercise exercise, @NotNull Set<TextSubmission> submissions) {
+            this.exercise = TextExerciseDTO.of(exercise);
+            this.submissions = submissions.stream().map(submission -> TextSubmissionDTO.of(exercise.getId(), submission)).toList();
+        }
+    }
+
+    private record ResponseDTO(String data) {
+    }
+
+    /**
+     * Calls the remote Athena service to submit a job for calculating automatic feedback
+     *
+     * @param exercise the exercise the automatic assessments should be calculated for
+     */
+    public void sendSubmissions(TextExercise exercise) {
+        sendSubmissions(exercise, 1);
+    }
+
+    /**
+     * Calls the remote Athena service to submit a Job for calculating automatic feedback
+     * Note: See `TextSubmissionService:getTextSubmissionsByExerciseId` for selection of Submissions.
+     *
+     * @param exercise   the exercise the automatic assessments should be calculated for
+     * @param maxRetries number of retries before the request will be canceled
+     */
+    public void sendSubmissions(TextExercise exercise, int maxRetries) {
+        if (!exercise.isFeedbackSuggestionsEnabled()) {
+            throw new IllegalArgumentException("The Exercise does not have feedback suggestions enabled.");
+        }
+
+        log.debug("Start Athena Submission Sending Service for Text Exercise '{}' (#{}).", exercise.getTitle(), exercise.getId());
+
+        // Find all text submissions for exercise (later we will support others)
+        Set<TextSubmission> textSubmissions = textSubmissionRepository.getTextSubmissionsWithTextBlocksByExerciseId(exercise.getId());
```

Comment:
Split this up into sub-requests and use Pageables to retrive submissions, we should not try to deal with 2000 object in memory at the same time.

Reply:
It makes sense to split it up 👍 Somehow, the old Athena worked despite always sending all submissions at once, but this solution is better - so I will change that.

Reply:
Added batching! 🎉